### PR TITLE
Feat/java

### PR DESCRIPTION
**Pull request by ccorsin**

### CODE REVIEW

File: public/actions/createPolicy.ts [hunk 1]
```diff
@@ -1,5 +1,21 @@
...
+//   },
+//   {
+//     country: [
+//       { name: "France", isHybridized: false },
+//       { name: "Spain", isHybridized: false },
+//       { name: "Germany", isHybridized: false },
+//     ],
+//   },
+// ];
+
```
Comment (Manuthor) [inline public/actions/createPolicy.ts:18]: should we keep this code?

File: public/actions/decryptDataInKms.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+//
+// Decrypt data in KMS
+//
+public static void decryptDataInKms(String kmsServerUrl, String apiKey, String userDecryptionKey, String ciphertext) {
+  KmsClient kmsClient = new KmsClient(kmsServerUrl, apiKey);
+  String plaintext = new String(
+    kmsClient.coverCryptDecrypt(userDecryptionKey, ciphertext)
+    .getPlaintext(),
```
Comment (Manuthor) [inline public/actions/decryptDataInKms.java:8]: EOL is maybe too much here

File: src/actions/createPolicy.ts [hunk 3]
```diff
@@ -1,5 +1,21 @@
...
+//   },
+//   {
+//     country: [
+//       { name: "France", isHybridized: false },
+//       { name: "Spain", isHybridized: false },
+//       { name: "Germany", isHybridized: false },
+//     ],
+//   },
+// ];
+
```
Comment (weblaetitia) [inline src/actions/createPolicy.ts:18]: Add this part below the function, and rename it `POLICY_AXIS_EXAMPLE`.

File: public/actions/createCovercryptKeyPair.java [hunk 4]
```diff
@@ -0,0 +1,8 @@
+//
+// Create Covercrypt Key Pair
+//
+public static void createCoverCryptMasterKeyPair(String kmsServerUrl, String apiKey) {
+  Policy policy = policy();
+  KmsClient kmsClient = new KmsClient(kmsServerUrl, apiKey);
```
Comment (Manuthor) [inline public/actions/createCovercryptKeyPair.java:6]: the actual `KmsClient` constructor is:
```
    public KmsClient(String server_url, Optional<String> api_key) {
        this(new RestClient(server_url, api_key));
    }
```

`apiKey` should be an `Optional<String>`

File: src/actions/defineCallbacks.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+// Define your callbacks
+//
+
+/**
+ * Fetch the Entry Table lines for the list of given {@link Uid32}. If a line does not exist, there should be no entry in the returned map.
+ *
+ * @param uids the unique {@link Uid32}s used as line id
+ * @return a {@link Map} of {@link Uid32} to {@link EntryTableValue}
+ * @throws CloudproofException if anything goes wrong
+ */
```
Comment (weblaetitia) [inline src/actions/defineCallbacks.java:11]: Those comments are really heavy. Do we really need them? In the JS part, there is no explanation of how callbacks work.

File: src/actions/encryptDataLocally.java [hunk 6]
```diff
@@ -0,0 +1,8 @@
+//
+// Encrypt data locally
+//
+public static void encryptDataLocally(String userDecryptionKey, String plaintext, Policy policy, MasterKeys masterKeys) {
```
Comment (weblaetitia) [inline src/actions/encryptDataLocally.java:4]: (sorry I'm not use to Java.) If Policy and MasterKeys are types: describe these types.

File: src/actions/createCovercryptKeyPair.java
Comment (Manuthor) [inline src/actions/createCovercryptKeyPair.java:1]: this file seems to be duplicated. Next files too.

**Review by weblaetitia — APPROVED** [review]

**Pull request merged**